File: apps/backend/go-service/routes/health.go
package routes

import (
	"net/http"
	"time"

	"github.com/gin-gonic/gin"
	"go-service/database"
)

// SetupHealthRoutes sets up health check routes
func SetupHealthRoutes(router *gin.Engine) {
	health := router.Group("/api/go/health")
	{
		health.GET("/", healthCheck)
		health.GET("/ready", readinessCheck)
		health.GET("/live", livenessCheck)
	}
}

// healthCheck handles basic health check
func healthCheck(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"status":    "healthy",
		"timestamp": time.Now().UTC().Format(time.RFC3339),
		"service":   "go-service",
	})
}

// readinessCheck handles readiness check with database connectivity
func readinessCheck(c *gin.Context) {
	checks := gin.H{
		"status":    "ready",
		"timestamp": time.Now().UTC().Format(time.RFC3339),
		"checks":    gin.H{},
	}

	// Check MySQL connection
	db := database.GetDB()
	if db != nil {
		sqlDB, err := db.DB()
		if err != nil {
			checks["checks"].(gin.H)["mysql"] = "error: " + err.Error()
			checks["status"] = "not ready"
		} else if err := sqlDB.Ping(); err != nil {
			checks["checks"].(gin.H)["mysql"] = "error: " + err.Error()
			checks["status"] = "not ready"
		} else {
			checks["checks"].(gin.H)["mysql"] = "connected"
		}
	} else {
		checks["checks"].(gin.H)["mysql"] = "not initialized"
		checks["status"] = "not ready"
	}

	statusCode := http.StatusOK
	if checks["status"] != "ready" {
		statusCode = http.StatusServiceUnavailable
	}

	c.JSON(statusCode, checks)
}

// livenessCheck handles liveness check
func livenessCheck(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"status":    "alive",
		"timestamp": time.Now().UTC().Format(time.RFC3339),
	})
}

File: apps/backend/python-service/app/routers/health.py
"""
Health check endpoints.
"""

from fastapi import APIRouter, HTTPException
from datetime import datetime
import asyncio

from app.database.mongodb import get_database
from app.database.postgres import engine

router = APIRouter()


@router.get("/")
async def health_check():
    """Basic health check."""
    return {
        "status": "healthy",
        "timestamp": datetime.utcnow().isoformat(),
        "service": "python-service"
    }


@router.get("/ready")
async def readiness_check():
    """Readiness check with database connectivity."""
    checks = {
        "status": "ready",
        "timestamp": datetime.utcnow().isoformat(),
        "checks": {}
    }

    # Check MongoDB
    try:
        db = get_database()
        if db is not None:
            await db.command('ping')
            checks["checks"]["mongodb"] = "connected"
        else:
            checks["checks"]["mongodb"] = "not initialized"
            checks["status"] = "not ready"
    except Exception as e:
        checks["checks"]["mongodb"] = f"error: {str(e)}"
        checks["status"] = "not ready"

    # Check PostgreSQL
    try:
        if engine:
            async with engine.connect() as conn:
                await conn.execute("SELECT 1")
            checks["checks"]["postgresql"] = "connected"
        else:
            checks["checks"]["postgresql"] = "not initialized"
            checks["status"] = "not ready"
    except Exception as e:
        checks["checks"]["postgresql"] = f"error: {str(e)}"
        checks["status"] = "not ready"

    if checks["status"] != "ready":
        raise HTTPException(status_code=503, detail=checks)

    return checks


@router.get("/live")
async def liveness_check():
    """Liveness check."""
    return {
        "status": "alive",
        "timestamp": datetime.utcnow().isoformat()
    }
